include/catch.hpp

```diff
@@ -144,6 +144,14 @@
 
 #define CATCH_ANON_TEST_CASE() INTERNAL_CATCH_TESTCASE()
 
+#ifdef CATCH_USE_STATIC_REQUIRE
+#define CATCH_STATIC_REQUIRE( expr )       static_assert( expr, #expr ); CATCH_SUCCEED( #expr )
```

Comment:
These macros need to be variadic to work properly when `expr` contains commas, e.g. `std::is_same<int, int>::value`, [see this](https://godbolt.org/z/V38BwI), specifically the fact that the last macro use causes compilation error, as it is parsed as a macro with two arguments, not just one.

Reply:
As noted in [the docs](https://github.com/catchorg/Catch2/blob/master/docs/assertions.md#expressions-with-commas), using an alias/typedef, or surrounded the `expr` with parentheses is what is required.

Now I would have put the extra parens in myself, but then the behaviour would be inconsistent with `REQUIRE` as it currently written.

You example would work with:

```cpp
STATIC_REQUIRE((std::is_same_v<int, int>));
```

which is how I am currently using it in my own work.

Reply:
The trick is that it doesn't have to be that way for single-argument macros:

```cpp
#include <type_traits>
#define MY_ASSERT2(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
MY_ASSERT2(std::is_same<int, int>::value);
```
[works](https://godbolt.org/z/uf7u6p) and uses the same trick `REQUIRE` does.